drgn/helpers/linux/tc.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) ByteDance, Inc. and its affiliates.
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Traffic Control (TC)
+--------------------
+
+The ``drgn.helpers.linux.tc`` module provides helpers for working with the
+Linux kernel Traffic Control (TC) subsystem.
+"""
+
+import operator
+
+from drgn import NULL, IntegerLike, Object
+from drgn.helpers.linux.list import hlist_for_each_entry
+
+__all__ = ("qdisc_lookup",)
+
+
+def qdisc_lookup(dev: Object, major: IntegerLike) -> Object:
+    """
+    Get a Qdisc from a device and a major handle number.  It is worth noting
+    that conventionally handles are hexadecimal, e.g. ``10:`` in a ``tc``
+    command means major handle 0x10.
+
+    :param dev: ``struct net_device *``
+    :param major: Qdisc major handle number.
+    :return: ``struct Qdisc *`` (``NULL`` if not found)
+    """
+    major = operator.index(major) << 16
+    roots = list()
+
+    roots.append(dev.qdisc)
```

Comment:
You can write this as:

```suggestion
    roots = [dev.qdisc]
```